engine.go

```diff
@@ -5,13 +5,12 @@ import (
 	"gorgonia.org/tensor"
 )
 
-type SymbolicEngine struct{}
```

Comment:
Did not find any reference anywhere, so I consider it as dead code.

Reply:
`SymbolicEngine` is for the upcoming v0.10.0 API